topology/probes/k8s/linker.go

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2018 IBM Corp.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package k8s
+
+import (
+	"github.com/skydive-project/skydive/graffiti/graph"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/probe"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AreLinked return true if (a, b) should be linked
+type AreLinked func(a, b interface{}) bool
+
+// ABLinker basis for a simple A to B linker
+type ABLinker struct {
+	Manager   string
+	Type      string
+	Graph     *graph.Graph
+	ACache    *ResourceCache
+	BCache    *ResourceCache
+	AreLinked AreLinked
+}
+
+// ABLinkerInterface returns the embbed ABLinker
+type ABLinkerInterface interface {
+	GetABLinker() *ABLinker
+}
+
+// GetABLinker returns inner object
+func (l *ABLinker) GetABLinker() *ABLinker {
+	return l
+}
+
+// NewEdgeMetadata create a new edge metadata
+func (l *ABLinker) NewEdgeMetadata() graph.Metadata {
```

Comment:
You already have too many "New...Metadata", please inline as it is used only once

Reply:
done